In [1]:
import analysis as an
import os
import glob
import os

In [2]:
basepath = '/Users/gracer/Google Drive/HCP/HCP_graph/1200/datasets/'

In [226]:
p = os.path.join(basepath,'tmp','6_*_dict*')
list_of_files = glob.glob(p) # * means all if need specific format then *.csv
latest_file = max(list_of_files, key=os.path.getctime)
print(latest_file)
submod_dict=an.onetoughjar(latest_file)


/Users/gracer/Google Drive/HCP/HCP_graph/1200/datasets/tmp/6_submod_dict_07-31-2020_06-05-09


In [230]:
submod_dict['no'][0]

centrality  clustering        PC  modules   zDegree   group
0     0.006490    0.643771  0.689564        0 -0.344401  normal
24    0.012522    0.607639  0.676096        0 -0.344401  normal
38    0.006565    0.547561  0.648773        0 -0.344401  normal
31    0.011106    0.610169  0.699115        0 -0.344401  normal

In [217]:
summary_dict['NR']['ob'].keys()

dict_keys(['mean_FC', 'graphs', 'clustering_coeff', 'btn_centrality', 'PC', 'modules'])

In [225]:
summary_dict['NR']['ob']['modules']['zdegree']

array([ 0.46594891,  1.04893508,  0.08219027,  1.21347393,  2.10582374,
        1.01422222, -0.11224895,  0.74041278,  0.06955671,  1.66453219,
        1.86191182, -0.60444668,  1.63846027,  1.03418729, -0.86139118,
       -0.16634163,  1.30934458,  0.17396251, -0.80391223, -0.0520899 ,
       -0.41284358, -0.36923382, -0.65414812,  0.17227346,  1.32701126,
        0.94183907,  0.09395972,  0.70357872, -1.14402472,  0.74506072,
       -0.19019402, -1.35177101,  0.71474002, -0.70005107,  0.10492573,
        1.03697314, -2.04475256,  0.68001898, -0.44118916,  0.88226656,
       -0.3377231 ,  0.11912269, -0.18823309, -0.64646582,  0.20077412,
        2.18814931, -0.52409036,  0.78266643,  1.08734743,  0.96086216,
        0.80419426,  0.54589679,  0.6693931 ,  1.2598472 , -1.4892002 ,
       -0.64317579, -0.50619175,  0.42997468, -0.32702492, -0.46411282,
       -0.82345222,  0.82187381,  1.51118263, -0.8264203 ,  1.28672116,
       -0.68527401, -0.17496798, -1.96455803, -0.20319132, -0.07

In [209]:
len(summary_dict['NR']['ob']['PC'][1])

100

In [103]:
def mod_world(dicti):
    mod_dict={}
    for key, value in dicti.items():
#         print(key)
        if key == 'modules':
#             print(value.keys())
            dicti[key]['Q']=community.modularity(value['partition'], value['graph'], weight='weight')

    edge_btw=nx.edge_betweenness_centrality(dicti['graphs'], normalized=True, weight='weight')
    dicti['edge_btw']=edge_btw
    nx.set_edge_attributes(dicti['graphs'], edge_btw, 'betweenness')

    unique, counts = np.unique(dicti['modules']['values'], return_counts=True)
    for i in unique:
        mod_dict.update({i:[]})
    for q, w in dicti['modules']['partition'].items():
        mod_dict[w].append(q)
    return(dicti, mod_dict)


In [190]:
[x,y]=mod_world(summary_dict['NR']['no'])

In [191]:
x['graphs'].edges(data=True)

EdgeDataView([(0, 0, {'weight': 1.0, 'betweenness': 0.0}), (0, 1, {'weight': 0.059682205128205136, 'betweenness': 0.0}), (0, 2, {'weight': 0.35860901923076927, 'betweenness': 0.0}), (0, 3, {'weight': -0.3289406025641025, 'betweenness': 0.7142424242424242}), (0, 4, {'weight': -0.028333147435897432, 'betweenness': 0.0}), (0, 5, {'weight': 0.055733461538461544, 'betweenness': 0.0}), (0, 6, {'weight': 0.01403624358974359, 'betweenness': 0.0}), (0, 7, {'weight': 0.01330883333333333, 'betweenness': 0.0}), (0, 8, {'weight': 0.111724, 'betweenness': 0.0}), (0, 9, {'weight': 0.0030129358974358995, 'betweenness': 0.0}), (0, 10, {'weight': -0.27599458974358976, 'betweenness': 0.009898989898989899}), (0, 11, {'weight': 0.0260046858974359, 'betweenness': 0.0}), (0, 12, {'weight': -0.12174789102564103, 'betweenness': 0.0}), (0, 13, {'weight': -0.004262429487179487, 'betweenness': 0.0}), (0, 14, {'weight': 0.16304087179487178, 'betweenness': 0.0}), (0, 15, {'weight': -0.17780821153846155, 'betweennes

In [192]:
y.keys()

dict_keys([0, 1, 2, 3, 4, 5, 6, 7])

In [193]:
g=summary_dict['NR']['no']

In [194]:
def sub_G(dicti, mod_dicti):
    subgraph_dict={}
    for key, value in mod_dicti.items():
        print(key)
        G=dicti['graphs']
        H = G.subgraph(value).copy()
        subgraph_dict[key]=H
    return(subgraph_dict)


In [195]:
subgraph_dict = sub_G(g,y)

0
1
2
3
4
5
6
7


In [196]:
g['graphs'].nodes(data=True)

NodeDataView({0: {'centrality': 0.0064897671177409555, 'clustering': 0.6437710437710438, 'PC': array([0.73282089, 0.38806599, 0.45853902, 0.1540596 , 0.60179288,
       0.57569998, 0.53532999, 0.26076176, 0.45376258, 0.63665278,
       0.14639205, 0.36155385, 0.23543609, 0.47300124, 0.7469558 ,
       0.28533176, 0.16167291, 0.64698378, 0.47912608, 0.38742065,
       0.28176444, 0.31499796, 0.42897714, 0.3085599 , 0.68205196,
       0.68608656, 0.36279311, 0.16774159, 0.59055189, 0.24204337,
       0.63951973, 0.69790872, 0.17439574, 0.41261648, 0.36380386,
       0.60384362, 0.4632951 , 0.38881714, 0.63907524, 0.46469497,
       0.28124348, 0.36077505, 0.28676497, 0.36790693, 0.19234107,
       0.40555609, 0.23161222, 0.31287778, 0.69232127, 0.42367493,
       0.4127796 , 0.46355675, 0.47913507, 0.38231477, 0.50502382,
       0.54735487, 0.49516657, 0.61258907, 0.57769781, 0.65128066,
       0.44733351, 0.62389881, 0.70879164, 0.78454273, 0.56531664,
       0.44355576, 0.76944213, 0.6

In [197]:
subgraph_dict[0].nodes(data=True)

NodeDataView({0: {'centrality': 0.0064897671177409555, 'clustering': 0.6437710437710438, 'PC': array([0.73282089, 0.38806599, 0.45853902, 0.1540596 , 0.60179288,
       0.57569998, 0.53532999, 0.26076176, 0.45376258, 0.63665278,
       0.14639205, 0.36155385, 0.23543609, 0.47300124, 0.7469558 ,
       0.28533176, 0.16167291, 0.64698378, 0.47912608, 0.38742065,
       0.28176444, 0.31499796, 0.42897714, 0.3085599 , 0.68205196,
       0.68608656, 0.36279311, 0.16774159, 0.59055189, 0.24204337,
       0.63951973, 0.69790872, 0.17439574, 0.41261648, 0.36380386,
       0.60384362, 0.4632951 , 0.38881714, 0.63907524, 0.46469497,
       0.28124348, 0.36077505, 0.28676497, 0.36790693, 0.19234107,
       0.40555609, 0.23161222, 0.31287778, 0.69232127, 0.42367493,
       0.4127796 , 0.46355675, 0.47913507, 0.38231477, 0.50502382,
       0.54735487, 0.49516657, 0.61258907, 0.57769781, 0.65128066,
       0.44733351, 0.62389881, 0.70879164, 0.78454273, 0.56531664,
       0.44355576, 0.76944213, 0.6

In [198]:
def df_maker(subgraph_dict, group):
    modstat_dict={}
    for mod, nodes in subgraph_dict.items():
        print(mod)
        modstat_dict[mod]={}
        for i in nodes:
            print(i)
            modstat_dict[mod].update({i:[]})
        modstat_dict[mod]=pd.DataFrame.from_dict(dict(subgraph_dict[mod].nodes(data=True)), orient='index')
        modstat_dict[mod]['group']=group
    return(modstat_dict)


In [199]:
modstat_dict=df_maker(subgraph_dict,'normal')

0
0
24
38
31
1
1
6
8
11
13
21
23
28
32
36
39
41
43
45
47
49
50
51
52
53
54
55
56
58
60
65
73
2
2
3
7
10
12
15
16
18
19
20
22
26
27
29
33
34
37
40
42
44
46
80
87
95
96
3
4
17
25
30
48
57
61
67
69
71
74
76
77
81
82
83
84
88
89
91
92
4
35
5
72
9
14
59
63
5
64
66
98
68
99
75
78
86
62
6
93
70
79
7
97
90
85
94


In [200]:
modstat_dict[7]

centrality  clustering  modules   zDegree   group
97    0.002353    0.581538        7 -0.567951  normal
90    0.002650    0.734505        7 -0.567951  normal
85    0.004103    0.699565        7 -0.567951  normal
94    0.003176    0.710453        7 -0.567951  normal

In [ ]:
summary_dict['NR']['no'].keys()

In [99]:
import pandas as pd
import community
import networkx as nx
import numpy as np

### Hubs  
intra-modular degree greater than 2.5   
a participant coefficient between 0 and 0.3 was considered a provincial hub   
between 0.3 and 0.75 was considered a connector hub  
between 0.75 and 1 was a kinless hub  
## non-hub 
ultra-peripheral node if the participation coefficient was between 0 and 0.05   
a peripheral node if the participation coefficient was between 0.05 and 0.62  
and a connector node if the participation coefficient was between 0.63 and 0.8  
kinless nodes if the participation coefficient was between 0.8 and 1   